src/lib/select/select.spec.ts

```diff
@@ -882,6 +895,94 @@ describe('MdSelect', () => {
 
     });
 
+    describe('limited space to open horizontally', () => {
+      beforeEach(async(() => {
+        select.style.position = 'absolute';
+        select.style.top = '200px';
+      }));
+
+      it('should stay within the viewport when overflowing on the left in ltr', async(() => {
+        select.style.left = '-100px';
+        trigger.click();
+        fixture.detectChanges();
+
+        fixture.whenStable().then(() => {
+          const panelLeft = document.querySelector('.mat-select-panel')
+              .getBoundingClientRect().left;
+          expect(panelLeft).toBeGreaterThan(0,
+              `Expected select panel to be inside the viewport in ltr.`);
+        });
+      }));
+
+      it('should stay within the viewport when overflowing on the right in rtl', async(() => {
+        dir.value = 'rtl';
+        select.style.left = '-100px';
```

Comment:
Seems like either the test name or the style here is out of sync. Did you mean to say "when overflowing on the left in rtl"?  The test on line 947 seems to cover the case on the right.